### PR TITLE
Match emojis by long name

### PR DESCRIPTION
**Pull request by lukebarnard1**
For example, searching for `:tong` will now match `:stuck_out_tongue` because the query `:tong` has all non-word characters removed, becoming `tong` and is then matched against the (long) names of emojis such as `smiling face with open mouth and tightly-closed eyes`.

Fixes https://github.com/vector-im/riot-web/issues/4620

### CODE REVIEW

**Comment by lukebarnard1** [comment]
@richvdh, another test timeout https://travis-ci.org/matrix-org/matrix-react-sdk/builds/255582505#L2074

File: src/autocomplete/EmojiProvider.js [hunk 1]
```diff
@@ -57,11 +59,13 @@ const EMOJI_SHORTNAMES = Object.keys(EmojiData).map((key) => EmojiData[key]).sor
         }
         return CATEGORY_ORDER.indexOf(a.category) - CATEGORY_ORDER.indexOf(b.category);
     },
-).map((a) => {
+).map((a, index) => {
     return {
         name: a.name,
         shortname: a.shortname,
         aliases_ascii: a.aliases_ascii ? a.aliases_ascii.join(' ') : '',
+        // Include the so that we can preserve the original order
```
Comment (dbkr) [inline src/autocomplete/EmojiProvider.js]: the what?

Comment (lukebarnard1) [inline src/autocomplete/EmojiProvider.js]: oops

**Comment by richvdh** [comment]
> @richvdh, another test timeout https://travis-ci.org/matrix-org/matrix-react-sdk/builds/255582505#L2074

Should be fixed by https://github.com/vector-im/riot-web/pull/4650

**Pull request merged**